src/main/java/gnosis/controller/GnosisController.java

```diff
@@ -0,0 +1,119 @@
+package gnosis.controller;
+
+import java.util.List;
+
+import gnosis.model.Command;
+import gnosis.model.Deadline;
+import gnosis.model.Event;
+import gnosis.model.Task;
+import gnosis.model.Todo;
+import gnosis.task.TaskCommandManager;
+import gnosis.task.TaskStorageManager;
+import gnosis.ui.GnosisUI;
+import gnosis.util.GnosisConstants;
+import gnosis.util.GnosisException;
+
+/**
+ * Represents Logic flow of interaction of user input and, to specified
+ * managers to perform from user command.
+ *
+ * @author Pawandeep Singh
+ * */
+public class GnosisController {
+
+    /** Task Manager to handle task commands */
+    private TaskCommandManager taskCommandManager;
+
+    /** UI view of Gnosis */
+    private GnosisUI view;
+
+    /**
+     * GnosisController constructor to connect with GnosisView and,
+     * to initialise TaskCommanagerManager with loaded task.
+     *
+     * @param view UI view to user.
+     */
+    public GnosisController(GnosisUI view) {
+        this.view = view;
+        this.taskCommandManager = new TaskCommandManager(TaskStorageManager.loadGnosisTasks());
+    }
+
+    /**
+     * loads Gnosis greeting message to UI.
+     *
+     */
+    public void loadGreetingMessage() {
+        view.displayGreetMessage(TaskStorageManager.isDataFileAvail());
+    }
+
+    /**
+     * Executes specified command from input.
+     *
+     * @param command Command specified by user.
+     * @param commandInput input to perform from command.
+     * @throws GnosisException If command not found.
+     */
+    public void executeCommand(String command, String commandInput) throws GnosisException {
+        Command gc;
+        try {
+            gc = Command.valueOf(command.toUpperCase().trim());
+        } catch (IllegalArgumentException e) {
+            throw new GnosisException(GnosisConstants.COMMAND_NOT_FOUND_MESSAGE);
+        }
+
+        // parse the commands
+        if (gc == Command.BYE) {
+            view.displayByeMessage();
+        } else {
+            // default is only task Manager, but future can include cases of other gnosis feautres, e.g phonebook
+            this.executeTaskCommand(gc, commandInput);
+        }
+    }
+
+    /**
+     * Executes specified command from input.
+     * Updates user tasks into file system.
+     *
+     * @param gc task command to perform
+     * @param taskInput input to perform from task command.
+     * @throws GnosisException If command not found.
+     * @throws NumberFormatException if taskInput for DONE and DELETE command cannot be converted to integer.
+     */
+    public void executeTaskCommand(Command gc, String taskInput) throws GnosisException, NumberFormatException {
+        switch (gc) {
```

Comment:
According to the coding standard, switch statements should have no indentation for the cases:

```java
switch(a) {
case "1":
case "2":
}

```

Reply:
Thanks for the lookout, I think I might have lost the convention when I merged the branches at some point. Thanks !